com.civicrm.hrjobroles/js/src/job-roles/services/date-validation.js

```diff
@@ -62,25 +92,34 @@ define([
              * @param {Date|string|int} start
              * @param {Date|string|int} end
              */
-            validate: function validate(start, end) {
-                if (start instanceof Date) {
-                    start = start.getTime();
-                }
-
-                if (end instanceof Date) {
-                    end = end.getTime();
-                }
+            validate: function validate(start, end, contract_start, contract_end) {
+                start = formatDate(start);
+                end = formatDate(end);
+                contract_start = formatDate(contract_start);
+                contract_end = formatDate(contract_end);
 
                 var start_date = moment(start, this.dateFormats, true);
                 var end_date = moment(end, this.dateFormats, true);
 
+                contract_start = moment(contract_start, this.dateFormats, true);
+                contract_end = moment(contract_end, this.dateFormats, true);
+
                 checkIfValuesAreValid(start_date, ['start_date']);
 
                 if (end === 0 || end) {
                     checkIfValuesAreValid(end_date, ['end_date']);
 
                     if (start) {
                         checkIfStartDateIsLower(start_date, end_date);
+                        checkIfContractStartDateIsLowerThanStart(contract_start, start_date);
+
+                        if (contract_end.isValid()) {
```

Comment:
If I'm not mistaken you could, just to keep the concerns separated, avoid checking if the *contract* dates (both start and end) are valid, since those should've already passed validation in the `hrjobcontract` app and as such they should be safely assumed as valid